libyul/backends/evm/StackLayoutGenerator.cpp

```diff
@@ -191,13 +256,16 @@ Stack createIdealLayout(Stack const& _operationOutput, Stack const& _post, Calla
 }
 }
 
-Stack StackLayoutGenerator::propagateStackThroughOperation(Stack _exitStack, CFG::Operation const& _operation)
+Stack StackLayoutGenerator::propagateStackThroughOperation(Stack _exitStack, CFG::Operation const& _operation, bool _aggressiveStackCompression)
 {
+	// Enable aggressive stack compression for recursive calls.
+	if (auto const* functionCall = get_if<CFG::FunctionCall>(&_operation.operation))
+		if (functionCall->recursive)
+			_aggressiveStackCompression = true;
```

Comment:
This part means that a call with `_aggressiveStackCompression = false` sometimes actually runs with `true` so trying to rerun with `true` in `propagateStackThroughBlock()` is pointless. Moving it to `propagateStackThroughBlock()` would let you avoid this.

Or is `createIdealLayout()` light enough that rerunning this is not an issue?

Reply:
This will only set ``_aggressiveStackCompression`` to true for the specific operation (if it's a recursive function call), while rerunning the entire block with ``true`` in ``propagateStackThroughBlock`` will attempt to rerun *every* operation in the block with aggressive stack compression.
So this is only pointless, if the entire block is only a single recursive function call (and recursive calls are expected to be rare anyways) - and rerunning this stuff is not overly expensive anyways, so I'd say it's fine.

Reply:
Oh, you're right. I missed the fact that it's rerunning the whole block, not just an operation.